106:  (hotfix) install newrelic for better application monitoring

bfreeds opened the pull request:
This PR adds the node agent for New Relic Application Performance Monitoring (APM).

This follows the github workflow for a [hotfix branch](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow).
![image](https://user-images.githubusercontent.com/5316367/75175771-f594af00-5700-11ea-8122-1934f49ac7c8.png)


The add-on has already been installed on the Heroku app instance.  This installs the node agent in `zap-api` and instantiates new relic during the application startup.  

It follows line-by-line the [New Relic Node installation instructions](https://rpm.newrelic.com/accounts/2631730/applications/setup#nodejs).

Addresses part of #106

julialucyhogan commented on queries/projects/show.sql:
```diff
@@ -143,43 +142,143 @@ SELECT
+        (CASE
+          WHEN mm.dcp_milestone = '780593bb-ecc2-e811-8156-1458d04d0698' THEN 58
+          ELSE mm.dcp_milestonesequence
+        END) AS display_sequence, -- note the sequence number is being overidden for this 'CPC Review of Modification Scope' milestone so that it will be inserted by date between the related city council review milestones
```
block quotes are possible in sql with /* comment */; might make this easier to read

hannahkates replied:
Hmm my syntax highlighting doesn't like that. I'm keeping the "--" but moving the comments to a separate line so it's easier to read

julialucyhogan commented on queries/projects/show.sql:
```diff
@@ -143,43 +142,143 @@ SELECT
+          WHEN mm.dcp_milestone = '8e3beec4-dad0-e711-8116-1458d04e2fb8' AND p.dcp_publicstatus <> 'Filed' THEN mm.dcp_actualenddate
+          WHEN mm.dcp_milestone = '780593bb-ecc2-e811-8156-1458d04d0698' THEN mm.dcp_actualenddate
+          ELSE NULL
+        END) AS display_date,
```
for these CASE statements that include non-trivial logic, i think it would be a good idea to comment them. Just something simple like: "For certain milestones, date used for display depends if the project has been filed"

julialucyhogan commented on queries/projects/show.sql:
```diff
@@ -143,43 +142,143 @@ SELECT
+          WHEN mm.dcp_milestone = '8e3beec4-dad0-e711-8116-1458d04e2fb8' THEN NULL
+          WHEN mm.dcp_milestone = '780593bb-ecc2-e811-8156-1458d04d0698' THEN NULL
+          ELSE NULL
+        END) AS display_date_2,
```
also would add a comment explaining what `display_date_2` is, or give it a more meaningful name

Commit:
106: Install new relic node agent

Commit:
106: Add newrelic configuration file

Copied from `node_modules/newrelic/newrelic.js` as per
step 3 of the newrelic node agent installation documentation
https://rpm.newrelic.com/accounts/2631730/applications/setup#nodejs

Note:  NEW_RELIC_LICENSE_KEY is set on Heroku config for zap-api.

Commit:
106: Initalize newrelic during nest application startup

Step 4 of the new relic node agent installation instructions
https://rpm.newrelic.com/accounts/2631730/applications/setup#nodejs

I used require instead of import because I wanted to follow the
installation docs and a forum post I saw that used require.

Using `import` would have caused an eslint error because I'd be importing
a variable that isn't used - I'm not sure if `require` behaves any
differently (by invoking the newrelic function call immediately).

bfreeds commented:
![image](https://user-images.githubusercontent.com/5316367/75176616-7c965700-5702-11ea-9614-9bf93df8c3f3.png)

allthesignals commented:
TODO: Fix the Circleci stuff
